tutorials/graphics/canvas2.C

```diff
@@ -82,7 +82,7 @@ void canvas2()
          hFrame->GetYaxis()->SetLabelOffset(0.02);
          hFrame->GetYaxis()->SetTitleFont(43);
          hFrame->GetYaxis()->SetTitleSize(16);
-         hFrame->GetYaxis()->SetTitleOffset(5);
+         hFrame->GetYaxis()->SetTitleOffset(1);
```

Comment:
The change in behaviour of TitleOffset I accept might be undesirable but I think sometimes its ok to accept behaviour changes if the new behaviour is better enough. Worth noting that this will only impact the absolute size fonts which aren't the default size anyway. So I would be surprised if it was an earthquake, rather a very minor tremor. 

But what is concerning for me is you say with my change to the code the titles get drawn "on the axis". For my change to have that effect wouldn't it have to mean that charheight would have to evaluate to 0 and so in your setup somehow the `gPad->GetAbsWNDC()` value is massive? As you can see from my screenshot above, I dont see titles appearing "on the axis" ... have you got screenshot to show what you mean? Any idea why it happens for you and not for me?

Reply:
Sorry I tried again and, yes, the axis titles are ok with your changes (both in TGAxis and in canvas2.C). I do not know why yesterday I end up with something wrong. I must have messed up something. But I found an other tutorial which does not work anymore with your change: https://root.cern/doc/master/ratioplotOld_8C.html . With your change the X axis title disappear in this example.

Reply:
Yeah so again that was just the TitleOffset needed changing from 4 to 1. 

Basically I think where previously you had to "fine tune" the title offset to account for your pad being a different size, with this new code you can leave everything as the default TitleOffset of 1. (I think 1 is the default, yes). 

How do you want to proceed? Personally if we agree that this change should go in then yes we can go through all the tutorial plots and probably just remove anywhere the TitleOffset is being specified because the pad being drawn is not full size and the font is a fixed size one. But its surely not worth that effort until we agree that we want to proceed with this "minor tremor" behaviour change.

What do you think?

Reply:
Yes your mod looks fine you should also change the tutorials I pointed. And also add a description of the fix in https://github.com/root-project/root/blob/master/README/ReleaseNotes/v626/index.md in the "2D Graphics Libraries" Section. You can also add you name on top.

Reply:
Ok thanks for giving the go-ahead on this one. However, one further change ...

In preparation for the description in the release notes I tried to think what we are trying to achieve here. It's best said to be "offsets for axis titles with absolute-sized fonts are now absolute too (i.e. no longer relative to pad size)". If you agree that is the goal, then I had to make one further tweak to the key line that's being changed, to:

```
charheight /= ((x1==x0) ? gPad->GetWw()*gPad->GetAbsWNDC() : gPad->GetWh()*gPad->GetAbsHNDC());
```

(I'll actually further protect this to prevent dividing by zero). The reason for this change is because I noticed with the previous line I had that while the x-axis behaves as expected (same distance regardless of pad height) the y-axis offset would grow and shrink as the window or pad was resized, which didn't feel right. So with the above line things look good.

Hope that'll be ok! ... You can test it once ive committed it with the fixes to the two tutorials we've mentioned